### PR TITLE
[wallet-ext] - transaction summary follow-up work

### PR DESCRIPTION
**Pull request by mamos-mysten**
## Description 

Related to: https://github.com/MystenLabs/sui/pull/11975

- Updates `balanceChangeSummary` and `objectChangeSummary` to be keyed by owner
- Copies some UI from Explorer (CoinStack) as well as UI changes 
- Fetches display data for objects to display in approval / receipt pages where available
- Cleans up some unnecessary logic 
- Updates Explorer to work with changes to summary data format

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: apps/core/src/hooks/useTransactionSummary.ts
Comment (mamos-mysten) [inline apps/core/src/hooks/useTransactionSummary.ts:19]: this diff is a little confusing but I just moved the `getSummary` body into the `useTransactionSummary` hook

File: apps/core/src/hooks/useMultiGetObjects.ts
Comment (mamos-mysten) [inline apps/core/src/hooks/useMultiGetObjects.ts:11]: had a hell of a time trying to type this in order allow options to be passed to `useQuery` 🥲

File: apps/wallet/src/ui/app/shared/transaction-summary/cards/CoinStack.tsx
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/shared/transaction-summary/cards/CoinStack.tsx:12]: Only use the fully qualified type to determine if we special case this. Symbols are not the right way to do it. If this is how the explorer works, please change this there too.

File: apps/wallet/src/ui/app/shared/transaction-summary/cards/CoinStack.tsx
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/shared/transaction-summary/cards/CoinStack.tsx:33]: Same here related to SUI special casing.

File: apps/wallet/src/ui/app/shared/transaction-summary/cards/CoinStack.tsx
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/shared/transaction-summary/cards/CoinStack.tsx:35]: And here

File: apps/wallet/src/ui/app/shared/transaction-summary/cards/CoinStack.tsx
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/shared/transaction-summary/cards/CoinStack.tsx:59]: I wonder if there's a better way to do the leading negative margin but not a big deal.

File: apps/wallet/src/ui/app/shared/transaction-summary/cards/CoinStack.tsx
Comment (Jordan-Mysten) [reply apps/wallet/src/ui/app/shared/transaction-summary/cards/CoinStack.tsx:35]: Also can you just move this to the `CoinIcon` special case?

File: apps/wallet/src/ui/app/shared/transaction-summary/cards/CoinStack.tsx
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/shared/transaction-summary/cards/CoinStack.tsx:18]: Alt should probably be the token name?

File: apps/wallet/src/ui/app/shared/transaction-summary/OwnerFooter.tsx
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/shared/transaction-summary/OwnerFooter.tsx:46]: If it's clickable it should be a `button`

Comment (Jordan-Mysten) [reply apps/wallet/src/ui/app/shared/transaction-summary/OwnerFooter.tsx:46]: Although this behavior seems really weird to me, why do we want this?

File: apps/wallet/src/ui/app/shared/transaction-summary/cards/ObjectChanges.tsx
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/shared/transaction-summary/cards/ObjectChanges.tsx:46]: As a small aside, is it possible to do this through property-checking instead? I actually want to deprecate `is` because I want to move to a world where folks property-check instead of doing it like this (because this depends on runtime schemas that are shipped with the SDK which I want to go away).

**Review by Jordan-Mysten — APPROVED** [review]
Some comments, a lot of questions, and things we should absolutely do before we merge.

My recommendation is to go through each of the comments and make sure they're either answered, addressed, or unimportant, then make a decision to merge when you feel confident in everything. Approving to unblock tho.

**Comment by mamos-mysten** [comment]
Current dependencies on/for this PR:
* main
  * **PR #12158** <a href="https://app.graphite.dev/github/pr/MystenLabs/sui/12158" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/MystenLabs/sui/12158?utm_source=stack-comment).

File: apps/core/src/utils/transaction/getObjectDisplayLookup.ts
Comment (mamos-mysten) [reply apps/core/src/utils/transaction/getObjectDisplayLookup.ts:10]: deleted `useGetObjectsDisplay`

**Pull request merged**
